src/main/java/io/github/thebusybiscuit/slimefun4/core/handlers/RainbowTickHandler.java

```diff
@@ -129,4 +157,38 @@ public boolean isSynchronized() {
         return true;
     }
 
+    private void setBed(Block b, Material material, BlockFace facing) {
+        switch (facing) {
+            default:
+            case NORTH:
+                setBedBlock(b, material, BlockFace.SOUTH, Part.FOOT, false);
+                setBedBlock(b.getRelative(0, 0, 1), material, BlockFace.SOUTH, Part.HEAD, true);
+                break;
+            case SOUTH:
+                setBedBlock(b.getRelative(0, 0, 1), material, BlockFace.NORTH, Part.FOOT, false);
+                setBedBlock(b, material, BlockFace.NORTH, Part.HEAD, true);
+                break;
+            case EAST:
+                setBedBlock(b, material, BlockFace.WEST, Part.FOOT, false);
+                setBedBlock(b.getRelative(1, 0, 0), material, BlockFace.WEST, Part.HEAD, true);
+                break;
+            case WEST:
+                setBedBlock(b.getRelative(1, 0, 0), material, BlockFace.EAST, Part.FOOT, false);
+                setBedBlock(b, material, BlockFace.EAST, Part.HEAD, true);
+                break;
+        }
+    }
+
+    private void setBedBlock(Block b, Material material, BlockFace facing, Part part, boolean doPhysics) {
+		b.setType(material, false);
+        BlockData data = b.getBlockData();
+        
+		if (data instanceof Bed) {
+			((Bed) data).setFacing(facing);
+            ((Bed) data).setPart(part);
+        }
+        
+        b.setBlockData(data, doPhysics);
```

Comment:
Indentation hell awaits you

Reply:
I dont know where you found that code but it looks perfectly fine on my end